data-service/src/test/java/com/amazonaws/blox/dataservice/integration/CreateEnvironmentIntegrationTest.java

```diff
@@ -0,0 +1,140 @@
+                .build());
+
+    // The environment is valid
+    checkEnvironmentValid(ENVIRONMENT_NAME, ACCOUNT_ID, CLUSTER_ONE, createEnvironmentResponse);
+  }
+
+  @Test
+  public void testCreateAnEnvironmentAlreadyExist() throws Exception {
+    final EnvironmentId createdEnvironmentId =
+        models
+            .environmentId()
+            .accountId(ACCOUNT_ID)
+            .environmentName(ENVIRONMENT_NAME)
+            .cluster(CLUSTER_ONE)
+            .build();
+    // I create an environment
+    dataService.createEnvironment(
+        models
+            .createEnvironmentRequest()
+            .taskDefinition(TASK_DEFINITION)
+            .environmentId(createdEnvironmentId)
+            .build());
+    thrown.expect(ResourceExistsException.class);
+    thrown.expectMessage(
+        String.format("environment with id %s already exists", createdEnvironmentId));
+
+    // I try to create another environment with the same name and cluster
+    dataService.createEnvironment(
+        models
+            .createEnvironmentRequest()
+            .taskDefinition(TASK_DEFINITION)
+            .environmentId(createdEnvironmentId)
+            .build());
+  }
+
+  @Test
+  public void testCreateTwoEnvironmentsWithTheSameNameButDifferentClusters() throws Exception {
+    final EnvironmentId createdEnvironmentId1 =
+        models
+            .environmentId()
+            .accountId(ACCOUNT_ID)
+            .environmentName(ENVIRONMENT_NAME)
+            .cluster(CLUSTER_ONE)
+            .build();
+    // I create an environment
+    final CreateEnvironmentResponse createEnvironmentResponse1 =
+        dataService.createEnvironment(
+            models
+                .createEnvironmentRequest()
+                .taskDefinition(TASK_DEFINITION)
+                .environmentId(createdEnvironmentId1)
+                .build());
+    // The environment is valid
+    checkEnvironmentValid(ENVIRONMENT_NAME, ACCOUNT_ID, CLUSTER_ONE, createEnvironmentResponse1);
+
+    final EnvironmentId createdEnvironmentId2 =
+        models
+            .environmentId()
+            .accountId(ACCOUNT_ID)
+            .environmentName(ENVIRONMENT_NAME)
+            .cluster(CLUSTER_TWO)
+            .build();
+    // I create anohter environment with the same name but different cluster
+    final CreateEnvironmentResponse createEnvironmentResponse2 =
+        dataService.createEnvironment(
+            models
+                .createEnvironmentRequest()
+                .taskDefinition(TASK_DEFINITION)
+                .environmentId(createdEnvironmentId2)
+                .build());
+
+    // The second environment is also valid
+    checkEnvironmentValid(ENVIRONMENT_NAME, ACCOUNT_ID, CLUSTER_TWO, createEnvironmentResponse2);
+  }
+
+  private void checkEnvironmentValid(
+      final String environmentName,
+      final String accountId,
+      final String cluster,
+      final CreateEnvironmentResponse createEnvironmentResponse) {
+    assertEquals(
```

Comment:
Better to use assertj, `assertThat(..).isEqualTo(..)`. It provides both type safe, easy to read and easy to use with fluent api.

Reply:
Sure

Reply:
And I'm willing to bet that that will make this readable enough to inline the `checkEnvironmentValid` method.